### PR TITLE
crate the save_model dir automatically if it does not exists

### PR DESCRIPTION
**Pull request by jerrychen1990**
solve issue：https://github.com/jina-ai/finetuner/issues/427

### CODE REVIEW

File: finetuner/tuner/callback/best_model_checkpoint.py [hunk 1]
```diff
@@ -44,6 +44,12 @@ def __init__(
         self._logger = logging.getLogger('finetuner.' + self.__class__.__name__)
         self._logger.setLevel(logging.INFO if verbose else logging.WARNING)
         self._save_dir = save_dir
+
+        # make sure the save_dir exists
+        if not os.path.exists(self._save_dir):
+            self._logger.info(f"{self._save_dir} not exists, will create one.")
```
Comment (gmastrapas) [inline finetuner/tuner/callback/best_model_checkpoint.py:50]: Use single instead of double quotes

**Review by gmastrapas — CHANGES_REQUESTED** [review]
Hey @jerrychen1990 ! Thanks for your contribution! Left a minor comment

**Comment by gmastrapas** [comment]
@jerrychen1990 you need to merge from main or rebase, and tests will probably be ok

**Comment by bwanglzu** [comment]
I have backed up the OSS version, we'll close this PR since we'll not working on the current main anymore.

**Pull request closed without merging**